### PR TITLE
Clean up Time, TimeDelta, and Slab to match modern style

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by kidder — DISMISSED** [review]

File: src/Time/Slab.cpp [hunk 1]
```diff
@@ -7,28 +7,78 @@
...
-// incomplete in the header.
+Slab::Slab(double start, double end) noexcept : start_(start), end_(end) {
+  ASSERT(start_ < end_, "Backwards Slab: " << start_ << " >= " << end_);
+}
+
+Slab Slab::with_duration_from_start(double start, double duration) noexcept {
+  return {start, start + duration};
+}
+
+Slab Slab::with_duration_to_end(double end, double duration) noexcept {
```
Comment (nilsvu) [inline src/Time/Slab.cpp]: These arguments should probably be `const` (also in the other functions)

File: src/Time/Time.cpp [hunk 2]
```diff
@@ -135,3 +314,16 @@ size_t hash<Time>::operator()(const Time& t) const noexcept {
   return boost::hash<Time>{}(t);
 }
 }  // namespace std
+
+// Miscellaneous other functions for TimeDelta
+
+TimeDelta abs(TimeDelta t) noexcept {
+  if (not t.is_positive()) {
+    t *= -1;
```
Comment (nilsvu) [inline src/Time/Time.cpp:322]: is this standard behavior to modify the argument in-place and also return it? Should there be a non-modifying version?

**Review by nilsvu — CHANGES_REQUESTED** [review]
please squash right away

File: src/Time/Time.cpp (hunk 2 above)
Comment (wthrowe) [reply src/Time/Time.cpp:322]: `t` isn't a reference.

Comment (nilsvu) [reply src/Time/Time.cpp:322]: oh right, nvm

**Comment by kidder** [comment]
@nilsleiffischer please look at

**Review by kidder — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]

**Comment by kidder** [comment]
ignoring random failure in travis

**Pull request merged**